instrumentation/go.mongodb.org/mongo-driver/v2/mongo/otelmongo/test/mongo_test.go

```diff
@@ -0,0 +1,262 @@
+		},
+		func(s sdktrace.ReadOnlySpan) bool {
+			return assert.Contains(t, s.Attributes(), attribute.String("db.operation", "insert"))
+		},
+		func(s sdktrace.ReadOnlySpan) bool {
+			return assert.Contains(t, s.Attributes(), attribute.String("db.mongodb.collection", "test-collection"))
+		},
+		func(s sdktrace.ReadOnlySpan) bool {
+			return assert.Equal(t, codes.Unset, s.Status().Code)
+		},
+	}
+
+	tt := []struct {
+		title          string
+		operation      func(context.Context, *mongo.Database) (interface{}, error)
+		mockResponses  []bson.D
+		excludeCommand bool
+		validators     []validator
+	}{
+		{
+			title: "insert",
+			operation: func(ctx context.Context, db *mongo.Database) (interface{}, error) {
+				return db.Collection("test-collection").InsertOne(ctx, bson.D{{Key: "test-item", Value: "test-value"}})
+			},
+			mockResponses:  []bson.D{{{Key: "ok", Value: 1}}},
+			excludeCommand: false,
+			validators: append(commonValidators, func(s sdktrace.ReadOnlySpan) bool {
+				for _, attr := range s.Attributes() {
+					if attr.Key == "db.statement" {
+						return assert.Contains(t, attr.Value.AsString(), `"test-item":"test-value"`)
+					}
+				}
+				return false
+			}),
+		},
+		{
+			title: "insert",
+			operation: func(ctx context.Context, db *mongo.Database) (interface{}, error) {
+				return db.Collection("test-collection").InsertOne(ctx, bson.D{{Key: "test-item", Value: "test-value"}})
+			},
+			mockResponses:  []bson.D{{{Key: "ok", Value: 1}}},
+			excludeCommand: true,
+			validators: append(commonValidators, func(s sdktrace.ReadOnlySpan) bool {
+				for _, attr := range s.Attributes() {
+					if attr.Key == "db.statement" {
+						return false
+					}
+				}
+				return true
+			}),
+		},
+	}
+	for _, tc := range tt {
+		tc := tc
+
+		title := tc.title
+		if tc.excludeCommand {
+			title = title + "/excludeCommand"
+		} else {
+			title = title + "/includeCommand"
+		}
+
+		t.Run(title, func(t *testing.T) {
+			md := drivertest.NewMockDeployment()
+
+			sr := tracetest.NewSpanRecorder()
+			provider := sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(sr))
+
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+			defer cancel()
+
+			ctx, span := provider.Tracer("test").Start(ctx, "mongodb-test")
+
+			addr := "mongodb://localhost:27017/?connect=direct"
+			opts := options.Client()
+			//nolint:staticcheck
+			opts.Deployment = md // This method is the current documented way to set the mongodb mock. See https://github.com/mongodb/mongo-go-driver/blob/v2.0.0/x/mongo/driver/drivertest/opmsg_deployment_test.go#L24
+			opts.Monitor = otelmongo.NewMonitor(
+				otelmongo.WithTracerProvider(provider),
+				otelmongo.WithCommandAttributeDisabled(tc.excludeCommand),
+			)
+			opts.ApplyURI(addr)
+
+			md.AddResponses(tc.mockResponses...)
+			client, err := mongo.Connect(opts)
+			if err != nil {
+				t.FailNow()
+			}
```

Comment:
Suggest using `require.NoError` for test-critical infrastructure:

```suggestion
			require.NoError(t, err)
```

Note: this will require updating the import path to include `"github.com/stretchr/testify/require"`